### DISCUSSION TITLE
Fabric and multiple models

### DISCUSSION THREAD

**Discussion by pfeatherstone** [Lightning-AI/pytorch-lightning#18368] [DDP / multi-GPU / multi-node]
When calling `fabric.setup()` there isn't an option to pass in multiple models. My current hack is to put them into an `nn.ModuleDict` but then the returned `_FabricModule` object isn't subscriptable like `nn.ModuleDict`. Is there a workaround? Note, the equivalent [function](https://huggingface.co/docs/accelerate/v0.21.0/en/package_reference/accelerator#accelerate.Accelerator.prepare) with [Huggingface Accelerate](https://huggingface.co/docs/accelerate/index) allows this.

**Comment by francotheengineer** [comment]
Have you tried this guide?: 
https://lightning.ai/docs/fabric/stable/advanced/multiple_setup.html#multiple-models-multiple-optimizers
You should be able to call `fabric.setup` multiple times, each with a new model, and optimiser if needed.

**Reply by pfeatherstone** [reply]
But then I need one optimiser per model...

**Reply by ringohoffman** [reply]
@pfeatherstone I think you should open an issue for better visibility. I have the exact same request.

**Comment by awaelchli** [comment]
Regarding the module dict, I opened an issue here #18427. I think it should be possible to support that on the `_FabricModule`. In the mean time, you can always access the original module via `model.module` and then use it like a dict/list.
